frontend/server/src/DAO/Users.php

```diff
@@ -447,4 +447,22 @@ final public static function validateDeletionToken(
 
         return boolval($count);
     }
+
+    public static function getUserDataByParentalToken(string $token): ?int {
+        $sql = 'SELECT
+                      u.user_id
```

Comment:
```suggestion
        $fields = join(
            ', ',
            array_map(
                fn (string $field): string => "u.{$field}",
                array_keys(
                    \OmegaUp\DAO\VO\Users::FIELD_NAMES
                )
            )
        );                   
        $sql = "SELECT
                      {$fields}
```

this way we can use the same technique used by `findByUsername` above to avoid two database roundtrips.

Reply:
ohh yes yes!